data/questions_gen_postgres.csv

```diff
@@ -7,33 +7,33 @@ What's the difference in time between the first and last paper published?,SELECT
-How does the ratio of publications to journals change over the years? Return the annual numbers of publications and journals as well.,"SELECT publication.year, COUNT(DISTINCT publication.pid) AS num_publications, COUNT(DISTINCT publication.jid) AS num_journals, CAST(COUNT(DISTINCT publication.pid) AS FLOAT) / NULLIF(COUNT(DISTINCT publication.jid), 0) AS ratio FROM publication GROUP BY publication.year ORDER BY publication.year;",academic,ratio,
+What is the ratio of publications presented in conferences to publications published in journals?,"SELECT CAST(COUNT(DISTINCT CASE WHEN cid IS NOT NULL THEN pid END) AS FLOAT) / NULLIF(COUNT(DISTINCT CASE WHEN jid IS NOT NULL THEN pid END), 0) AS ratio FROM publication;",academic,ratio,
+What is the ratio of the total number of publications to the total number of keywords within each domain ID? Show all domain IDs.,"SELECT domain_publication.did, CAST(COUNT(DISTINCT domain_publication.pid) AS FLOAT) / NULLIF(COUNT(DISTINCT domain_keyword.kid), 0) AS publication_to_keyword_ratio FROM domain_publication LEFT JOIN domain_keyword ON domain_publication.did = domain_keyword.did GROUP BY domain_publication.did ORDER BY publication_to_keyword_ratio DESC NULLS LAST;SELECT domain_publication.did, CAST(COUNT(DISTINCT domain_publication.pid) AS FLOAT) / NULLIF(COUNT(DISTINCT domain_keyword.kid), 0) AS publication_to_keyword_ratio FROM domain_keyword LEFT JOIN domain_publication ON domain_publication.did = domain_keyword.did GROUP BY domain_publication.did ORDER BY publication_to_keyword_ratio DESC NULLS LAST;SELECT d.did, COALESCE(CAST(COUNT(DISTINCT dp.pid) AS FLOAT) / NULLIF(COUNT(DISTINCT dk.kid), 0), 0) AS publication_to_keyword_ratio FROM domain d LEFT JOIN domain_publication dp ON d.did = dp.did LEFT JOIN domain_keyword dk ON d.did = dk.did GROUP BY d.did ORDER BY publication_to_keyword_ratio DESC NULLS LAST;",academic,ratio,
+How does the ratio of publications to journals change over the years? Return the annual numbers of publications and journals as well.,"SELECT p.year, COUNT(DISTINCT p.pid) AS num_publications, COUNT(DISTINCT j.jid) AS num_journals, CAST(COUNT(DISTINCT p.pid) AS FLOAT) / NULLIF(COUNT(DISTINCT j.jid), 0) AS ratio FROM publication p LEFT JOIN journal j ON p.jid = j.jid GROUP BY p.year ORDER BY p.year;",academic,ratio,

```

Comment:
I ran both the queries before and after and they do return the same result - just to confirm that it's expected? I also realised that `jid` in `journal` but not in `publication` won't able to show up here as it wouldn't have the year column from `publication`.

Reply:
Oh yes. That's true yea. So in effect it's the same as the original query without the JOIN. Thanks for pointing this out. Then I'll just revert to the old one.